pkg/controller/crd_watcher.go

```diff
@@ -342,3 +374,60 @@ func NewCRDWatcher(crdManager ctrl.Manager, kind model.SubscribeKind, crdGenerat
 		sendDataHandler:      sendDataHandler,
 	}
 }
+
+func NewActiveCRDWatcher(crdManager ctrl.Manager, kind model.SubscribeKind, crdGenerator func() client.Object) *ActiveCRDWatcher {
+	return &ActiveCRDWatcher{
+		kind:         kind,
+		Client:       crdManager.GetClient(),
+		logger:       ctrl.Log.WithName("controller").WithName(kind),
+		scheme:       crdManager.GetScheme(),
+		crdGenerator: crdGenerator,
+	}
+}
+
+func (a *ActiveCRDWatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	crd := a.crdGenerator()
+	if err := a.Get(ctx, req.NamespacedName, crd); err != nil {
+		k8sApiErr, ok := err.(*k8sApiError.StatusError)
+		if !ok {
+			return ctrl.Result{
+				Requeue:      false,
+				RequeueAfter: 0,
+			}, nil
+		}
+		if k8sApiErr.Status().Code != http.StatusNotFound {
+			return ctrl.Result{
+				Requeue:      false,
+				RequeueAfter: 0,
+			}, nil
+		}
+
+		// cr had been deleted
+		crd = nil
+	}
+	switch a.kind {
+	case TrafficRouterKind:
+		if crd != nil {
+			cls := crd.(*traffic.TrafficRouter)
+			_, err := k8sclient.ApplyCRD(ctx, cls.Namespace, cls.Name, transform.BuildUnstructuredVirtualService(cls), gvr.VirtualServiceGVR)
```

Comment:
k8sclient is a client instanced from k8s.io/client-go, and there is a client in CRDWatcher too, so maybe can try to replace k8sclient by the client in CRDWatcher. 

This is only my opinion ~

Reply:
I will try to reuse the k8s-client in CRDWatcher! Thanks